packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -70,14 +72,17 @@ export const UploadControls = (): JSX.Element => {
     items: items! as FileItem[],
   });
 
-  return items ? (
+  return items && items.length > 0 ? (
     <>
       <Title />
+      <AddFiles disable={disable} />
+      <AddFolder disable={disable} />
```

Comment:
Since `useHandleUpload` doesn't update global state, I made a local state variable and passed it to these components as props. Does anyone have thoughts on this?

Reply:
Think you can just iterate over the `tasks` and check whether they all have completed to infer `loading`?